### PR TITLE
request-cert: vendor deps, migrate to latest K8S client

### PR DESCRIPTION
**Pull request by bharatnc**
This PR introduces `go mod` for vendoring dependencies in this repo.

- Start by running `go mod init` to init mod files.
- Then run `go mod vendor` to fetch and vendor the deps.

`vendor` folder is not commited. But to verify just need to run the
above steps.

Also migrated the K8S API clients in multiple places to V1 version.

Issues: #26 and #27

### CODE REVIEW

**Comment by bharatnc** [comment]
@joshimhoff created this single PR for issues #26 and #27 

Vendoring deps and updating the client libraries are very much dependent on each other so it was easier this way. Let me know if this looks good to yu=ou.

File: request-cert/k8s_certificates.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ import (
 	"time"
 
 	"github.com/pkg/errors"
+
+	"golang.org/x/tools/go/ssa/interp/testdata/src/os"
```
Comment (joshimhoff) [inline request-cert/k8s_certificates.go]: I don't see this being used anywhere?

File: request-cert/k8s_certificates.go [hunk 2]
```diff
@@ -22,6 +22,9 @@ import (
 	"time"
 
 	"github.com/pkg/errors"
+
```
Comment (joshimhoff) [inline request-cert/k8s_certificates.go]: Kill whitespace here and on line 27.

File: request-cert/k8s_certificates.go [hunk 3]
```diff
@@ -22,6 +22,9 @@ import (
 	"time"
```
Comment (joshimhoff) [inline request-cert/k8s_certificates.go]: How did you test? Ugh no CI in this repo!

**Review by joshimhoff — COMMENTED** [review]
Thanks for doing this.

File: request-cert/k8s_certificates.go (hunk 3 above)
Comment (joshimhoff) [reply request-cert/k8s_certificates.go]: Or makefile...

File: request-cert/k8s_certificates.go (hunk 1 above)
Comment (bharatnc) [reply request-cert/k8s_certificates.go]: nice catch, removed.

File: request-cert/k8s_certificates.go (hunk 2 above)
Comment (bharatnc) [reply request-cert/k8s_certificates.go]: It's actually a lspace b/w different groups of imports. Std imports, github/* and then k8s.io/* in this case. I can remove if necessary.

File: request-cert/k8s_certificates.go (hunk 3 above)
Comment (bharatnc) [reply request-cert/k8s_certificates.go]: Yea, no CI or Makefile. I can add Makefile though,. Maybe I can test locally.

**Pull request closed without merging**